src/ipz_parser.cpp

```diff
@@ -625,4 +625,198 @@ types::DbusVariantType IpzVpdParser::readKeywordFromHardware(
+                                 m_vpdVector.end());
+        }
+        else
+        {
+            l_kwdDataLength = *l_iterator;
+            std::ranges::advance(l_iterator, sizeof(types::KwSize),
+                                 m_vpdVector.end());
+        }
+
+        if (l_kwName == i_keywordName)
+        {
+            // Before writing the keyword's value, get the maximum size that can
+            // be updated.
+            const auto l_lengthToUpdate =
+                i_keywordData.size() <= l_kwdDataLength ? i_keywordData.size()
+                                                        : l_kwdDataLength;
+
+            // Set the keyword's value on vector. This is required to update the
+            // record's ECC based on the new value set.
+            const auto i_keywordDataEnd = std::ranges::next(
+                i_keywordData.cbegin(), l_lengthToUpdate, i_keywordData.cend());
+
+            std::copy(i_keywordData.cbegin(), i_keywordDataEnd, l_iterator);
+
+            // Set the keyword's value on hardware
+            const auto l_kwdDataOffset = std::distance(m_vpdVector.begin(),
+                                                       l_iterator);
+            m_vpdFileStream.seekp(m_vpdStartOffset + l_kwdDataOffset,
+                                  std::ios::beg);
+
+            std::copy(i_keywordData.cbegin(), i_keywordDataEnd,
+                      std::ostreambuf_iterator<char>(m_vpdFileStream));
+
+            // return no of bytes set
+            return l_lengthToUpdate;
+        }
+
+        // next keyword search
+        std::ranges::advance(l_iterator, l_kwdDataLength, m_vpdVector.end());
+
+        // next keyword name
+        l_kwName = std::string(
+            l_iterator,
+            std::ranges::next(l_iterator, Length::KW_NAME, m_vpdVector.end()));
+    }
+
+    // Keyword not found
+    throw std::runtime_error("Given keyword not found.");
+}
+
+int IpzVpdParser::writeKeywordOnHardware(
+    const types::WriteVpdParams i_paramsToWriteData)
+{
+    // Extract record, keyword and value from i_paramsToWriteData
+    types::Record l_recordName;
+    types::Keyword l_keywordName;
+    types::BinaryVector l_keywordData;
+
+    if (const types::IpzData* l_ipzData =
+            std::get_if<types::IpzData>(&i_paramsToWriteData))
+    {
+        l_recordName = std::get<0>(*l_ipzData);
+        l_keywordName = std::get<1>(*l_ipzData);
+        l_keywordData = std::get<2>(*l_ipzData);
+    }
+    else
+    {
+        logging::logMessage(
+            "Input parameter type provided isn't compatible with the given VPD type.");
+        throw types::DbusInvalidArgument();
+    }
+
+    if (l_recordName == "VHDR" || l_recordName == "VTOC")
+    {
+        logging::logMessage("Write operation not allowed on the given record.");
+
+        // Zero bytes updated.
+        return 0;
+    }
+
+    if (l_keywordData.size() == 0)
+    {
+        logging::logMessage("Given keyword's data length is 0.");
```

Comment:
we can follow this -

on successful execution -  return number of bytes updated.
on failure - throw exception - so that we don't suppress the specific exception here and caller will get the right exception type to handle.

Reply:
Ok.